### PR TITLE
Swagger spec for Server DNS Aliasing

### PR DESCRIPTION
**Pull request by maboja-msft**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by maboja-msft** [comment]
@jaredmoo @nathannfan Please comment

File: specification/sql/resource-manager/readme.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ input-file:
 - Microsoft.Sql/2015-05-01-preview/syncMembers.json
 - Microsoft.Sql/2015-05-01-preview/virtualNetworkRules.json
 - Microsoft.Sql/2017-03-01-preview/cancelOperations.json
+- Microsoft.Sql/2017-03-01-preview/serverDnsAliases.json
```
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: please also add to schema-2017-03-preview section

File: specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/examples/ServerDnsAliasAcquire.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿{
+  "parameters": {
+    "subscriptionId": "00000000-1111-2222-3333-444444444444",
+    "resourceGroupName": "Default",
+    "serverName": "dns-alias-new-server",
+    "dnsAliasName": "dns-alias-name-1",
+    "api-version": "2017-03-01-preview",
+	"parameters": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/examples/ServerDnsAliasAcquire.json]: nit: tabs/spaces here and in other example files

File: specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/serverDnsAliases.json [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+            "description": "The name of the server dns alias.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/serverDnsAliases.json]: is there no body parameter?

File: specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/serverDnsAliases.json [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ServerDnsAliasAcquisition": {
+      "description": "A server dns alias acquisition request.",
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/serverDnsAliases.json]: `DNS` or `dns`? Please pick one and stuck with it :)

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Overall looks great, minor comments. Please confirm that the service is in at least 1 production region and you have .net sdk tests passing that use code generated from this swagger.

File: specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/serverDnsAliases.json (hunk 3 above)
Comment (maboja-msft) [reply specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/serverDnsAliases.json]: No, those requests don't have body

File: specification/sql/resource-manager/readme.md (hunk 1 above)
Comment (maboja-msft) [reply specification/sql/resource-manager/readme.md]: This tag does not exist, so I have created it

**Comment by salameer** [comment]
Captain @alvadb 

Ping on this one :)

**Comment by maboja-msft** [comment]
In response to Jared's comment: yes, service is in at least one production region (in fact it is deployed and working in all regions) and .net client tests are passing.

**Review by alvadb — APPROVED** [review]
Please fix example errors in next commit.

**Pull request merged**